frontend/taipy-gui/src/components/Taipy/FileSelector.spec.tsx

```diff
@@ -98,4 +105,180 @@ describe("FileSelector Component", () => {
                 },
             });
     });
+    it("handles drag over event", () => {
+        const { getByRole } = render(<FileSelector label="FileSelectorDrag" />);
+        const fileSelector = getByRole("button");
+
+        // Create a mock DragEvent and add a dataTransfer object
+        const mockEvent = new Event("dragover", { bubbles: true }) as unknown as DragEvent;
+        Object.assign(mockEvent, {
+            dataTransfer: {
+                dropEffect: "",
+            },
+        });
+
+        // Dispatch the mock event
+        fireEvent(fileSelector, mockEvent);
+
+        // Add assertion to check if dropEffect is set to "copy"
+        expect(mockEvent.dataTransfer!.dropEffect).toBe("copy");
+    });
+
+    it("handles file drop", async () => {
+        // Create a mock file
+        const file = new File(["(⌐□_□)"], "chucknorris.png", { type: "image/png" });
+
+        const { getByRole } = render(<FileSelector label="FileSelectorDrop" />);
+        const elt = getByRole("button");
+        // Create a mock DragEvent and add the mock file to the event's dataTransfer.files property
+        const mockEvent = new Event("drop", { bubbles: true }) as unknown as DragEvent;
+        Object.assign(mockEvent, {
+            dataTransfer: {
+                files: [file],
+            },
+        });
+        // Dispatch the mock event
+        fireEvent(elt, mockEvent);
+        expect(uploadFile).toHaveBeenCalledTimes(1);
+    });
+
+    it("resets dropLabel and dropSx on drag leave", async () => {
+        const { getByRole, getByTestId } = render(<FileSelector />);
+        const elt = getByRole("button");
+
+        // Create a mock DragEvent
+        const mockEvent = new Event("dragleave", { bubbles: true }) as unknown as DragEvent;
+
+        // Dispatch the mock event
+        fireEvent(elt, mockEvent);
+
+        // Add assertions to check if dropLabel and dropSx have been reset
+        const dropLabelElement = getByTestId("file-selector");
```

Comment:
can't you get this element with the gettext(label) (default of specific) ?

Reply:
resolved